workflow/controller/operator.go

```diff
@@ -450,11 +463,32 @@ func (woc *wfOperationCtx) podReconciliation() error {
 	}
 
 	for _, pod := range podList.Items {
+		origNodeStatus := *woc.wf.Status.DeepCopy()
 		performAssessment(&pod)
 		err = woc.applyExecutionControl(&pod)
 		if err != nil {
 			woc.log.Warnf("Failed to apply execution control to pod %s", pod.Name)
 		}
+		err = woc.checkAndCompress()
+		if err != nil {
+			woc.wf.Status = origNodeStatus
+			nodeNameForPod := pod.Annotations[common.AnnotationKeyNodeName]
+			nodeID := woc.wf.NodeID(nodeNameForPod)
+			node := woc.wf.Status.Nodes[nodeID]
+			node.Message = fmt.Sprintf("%v", err)
+			woc.log.Warn(node.Message)
+			node.Outputs = nil
+			node.FinishedAt = metav1.Time{Time: time.Now().UTC()}
+			node.Phase = wfv1.NodeError
```

Comment:
There is a helper for all of this: `woc.markNodePhase()`. Can you use that instead?

Reply:
fixed